odl/solvers/nonsmooth/proximal_operators.py

```diff
@@ -839,22 +861,34 @@ def __init__(self, sigma):
 
             Parameters
             ----------
-            sigma : positive float
-                Step size parameter
+            sigma : positive float and componentwise positive space.element
+                Step size parameter. If not a float, then it is assumed to be
+                a vector which contains stepsizes for each component.
             """
             super(ProximalConvexConjL2Squared, self).__init__(
                 domain=space, range=space, linear=g is None)
-            self.sigma = float(sigma)
+            if np.isscalar(sigma):
+                self.sigma = float(sigma)
+            else:
+                self.sigma = space.element(sigma)
 
         def _call(self, x, out):
             """Apply the operator to ``x`` and store the result in ``out``"""
             # (x - sig*g) / (1 + sig/(2 lam))
             sig = self.sigma
-            if g is None:
-                out.lincomb(1.0 / (1 + 0.5 * sig / lam), x)
-            else:
-                out.lincomb(1.0 / (1 + 0.5 * sig / lam), x,
-                            -sig / (1 + 0.5 * sig / lam), g)
+            if np.isscalar(sig):
+                if g is None:
+                    out.lincomb(1.0 / (1 + 0.5 * sig / lam), x)
+                else:
+                    out.lincomb(1.0 / (1 + 0.5 * sig / lam), x,
+                                -sig / (1 + 0.5 * sig / lam), g)
+            elif sig in space:
+                if g is None:
+                    x.divide(1 + 0.5 / lam * sig, out=out)
+                else:
+                    sig.multiply(g, out=out)
+                    out.lincomb(1.0, x, -1.0, out=out)
+                    out.divide(1 + 0.5 / lam * sig, out=out)
 
```

Comment:
Else? The rule is in general that if you have an "else" case that should never happen, you add `RuntimeException('This should not occur')` or something like it, otherwise this can give silent errors